app/src/userpages/components/StreamPage/Edit/ConfigureView/index.jsx

```diff
@@ -1,247 +1,223 @@
-                                                    options={this.typeOptions}
-                                                    value={this.typeOptions.find((t) => t.value === field.type)}
-                                                    onChange={(o) => this.onFieldTypeChange(field.name, o.value)}
-                                                />
-                                            </SplitControl>
-                                        </FieldItem>
-                                    </div>
-                                </div>
-                            ))}
-                        </FieldList>
-                    </Fragment>}
-                {!isAddingField &&
-                    <React.Fragment>
-                        <Button
-                            kind="secondary"
-                            className={styles.addFieldButton}
-                            onClick={this.addNewField}
-                            disabled={disabled}
-                        >
-                            <Translate value="userpages.streams.edit.configure.addField" />
-                        </Button>
-                        <Button
-                            kind="secondary"
-                            outline
-                            onClick={this.autodetectFields}
-                            disabled={this.props.fieldsAutodetectFetching || disabled}
-                        >
-                            {!this.props.fieldsAutodetectFetching && (
-                                <Translate value="userpages.streams.edit.configure.autodetect" />
-                            )}
-                            {this.props.fieldsAutodetectFetching && (
-                                <Fragment>
-                                    <Translate value="userpages.streams.edit.configure.waiting" />
-                                    <Spinner size="small" className={styles.spinner} color="white" />
-                                </Fragment>
-                            )}
-                        </Button>
-                    </React.Fragment>
-                }
-                {isAddingField &&
-                    <NewFieldEditor
-                        previousFields={this.getStreamFields()}
-                        onConfirm={this.confirmAddField}
-                        onCancel={this.cancelAddField}
-                    />
-                }
-            </div>
-        )
-    }
-}
+        updateStream('config.fields', fields)
+    }, [streamFields, updateStream])
+
+    const streamId = stream.id
+    const autodetectFields = useCallback(async () => {
+        if (streamId) {
+            try {
+                await dispatch(streamFieldsAutodetect(streamId))
 
-const mapStateToProps = (state: StoreState): StateProps => ({
-    stream: selectEditedStream(state),
-    fieldsAutodetectFetching: selectFieldsAutodetectFetching(state),
-})
+                Notification.push({
```

Comment:
I wonder if we should show notifications in we're not mounted anymore…

Reply:
Should not, added the missing translation as well